### ISSUE TITLE
No timeout on request

### ISSUE THREAD

**Issue by duraki** [robacarp/icmp.cr#1]
Cool library, I enjoy using it.

When sending request on dead (aka, ping failed) address, the command just hangs.

https://github.com/robacarp/icmp.cr/blob/master/src/icmp/request.cr

```crystal
require "icmp"

HOST = "227.227.227.217"
ICMP::Ping.new(HOST).ping.["success"]
ICMP::Ping.new(HOST).ping(count:1)["success"]
```

Yielding it does not fix the issue.

Cheers

**Comment by robacarp** [comment]
@duraki thanks for looking and sorry for the late response. Somehow I turned off watch on this repo and didn't get notified when you posted.

I believe this is a bug.

**Comment by robacarp** [comment]
fixed by #4
